Location.java

```diff
@@ -1,3 +1,93 @@
+import java.util.ArrayList;
+import java.util.Iterator;
+
 public class Location {
+    
+    private String locationName;
+    private String locationDescription;
+    private ArrayList<Item> itemList;
+
+public Location(String pName, String pDescription){
+    locationName = pName;
+    locationDescription = pDescription;
+    itemList = new ArrayList<Item>();
+}
+
+public String getName(){
+    return locationName;
+}
+
+public String getDescription(){
+    return locationDescription;
+}
 
+public void setName(String pName){
+    locationName = pName;
 }
+
+public void setDescription(String pDescription){
+    locationDescription = pDescription;
+}
+
+public void addItem (Item itemToAdd){
+    itemList.add(itemToAdd);
+}
+
+public boolean hasItem (String itemSearching){
+    Iterator<Item> iter = itemList.iterator();
+    while (iter.hasNext()){
+        Item nextTo = iter.next();
+        String itemName = nextTo.getName();
+        if(itemSearching.equalsIgnoreCase(itemName)){
+            return true;
+        }
+        iter.hasNext();
+    }
+    return false;
+}
+
+public Item getItem(String itemGetting){
+    Iterator<Item> iter = itemList.iterator();
+    while (iter.hasNext()){
+        Item nextTo = iter.next();
+        String itemName = nextTo.getName();
+        if(itemGetting.equalsIgnoreCase(itemName)){
+            return nextTo;
+        }
+        iter.hasNext();
+    }
+    return null;
+    }
+
+public Item getItem(int index){
+    if(index<0 || index>itemList.size()){
```

Comment:
what if index == itemList.size()? 
can this still access items?

Reply:
I changed it to >=

Reply:
But here is another thing I am not sure how to write it. So 
1. if index is negative for >= size, return null: done 
2. run through arraylist, if there is anything match, return that: done. 
3. run through arraylist, found nothing, return null: here is where I don't know how to write. Do you have any idea?

Reply:
I think you are confused with the getItem(String) method. In this case, you "will" find the item no matter what because you are checking it with index. Once the index is less than array size and not negative, then you know that index will have something else.